### ISSUE TITLE
Current Player Device Wrong

### ISSUE THREAD

**Issue by nunofmds** [custom-cards/spotify-card#116]
**Describe the bug**

* [X ] Reproducable on a plain lovelace panel without card, style or theme mods
* [X ] Using latest version of spotcast and spotify-card [ v2.3.3]
* [X ] Using latest stable version of Home Assistant
* [X ] I have Spotify Premium
* [X ] I have setup the Spotify integration in Home Assistant

**To Reproduce**
1. Go to lovelace dashboard, select a group player with several cast devices
2. Click on select any playlist and press play
3. The current playing device changes to a standalone device from the group instead of keep the selection. The play continues to work on all of the group, so the functional part is working, only the selected device or current playing changes.

**Expected behavior**
Keep the current playing device, group and any other that's selected.

**Screenshots**
Selected to play on Ceiling All

![image](https://user-images.githubusercontent.com/8095169/96489721-b723a580-1237-11eb-8136-a77b4f3f3e8a.png)

Shows playing on Upstairs Bathroom but the playing is on Ceiling All
![image](https://user-images.githubusercontent.com/8095169/96489738-bf7be080-1237-11eb-8e62-ca4a1caad2e6.png)

**Environment (please complete the following information):**
 - OS: Docker
 - Browser: Chrome, IE, Android App
 - Version: All latest!
 - HA version: Latest
 - spotcast version: Latest

**Configuration**
Nothing special, everything standard.

**Additional context**

type: 'custom:spotify-card'
limit: 8
spotify_entity: media_player.spotify_xnuno
account: default
grid_covers_per_row: 4
display_style: grid
playlist_type: discover-weekly

**Comment by FL550** [comment]
If you open the spotify app, what is the playback device shown there? 
The playback device is taken directly from spotify, so this could be the issue.

**Comment by nunofmds** [comment]
Yes it matches the info that's on the lovelace card... but I started the playlist from the lovelace card and selected the group and not the device currently being displayed on both, spotify and card.

**Comment by FL550** [comment]
The card uses spotify for all information about playback. 
It can start playback on chromecast devices itself but any further interaction is done directly with spotify. So as far as I can see, there is unfortunately nothing we can do about this issue.

**Comment by nunofmds** [comment]
Yes but if I start the playlist from spotify and select the group it shows up correct on both, spotify and card.

This seems to be an issue from the card itself and the way it starts the broadcast or something...

**Comment by FL550** [comment]
Maybe @fondberg has an idea?

**Comment by fondberg** [comment]
I think this is an issue with groups only and the way spotify reports playback. Chromecast has implemented groups but spotify hasnt so there is a master cast device which is listed as playing in spotify apis if i remember correctly

**Comment by nunofmds** [comment]
Ok... strange that if i start the cast from spotify app everything works as expected on both sides... anything I can do to help fixing this?

**Comment by fondberg** [comment]
I need to try an replicate it. Will have some time soon to look into it.

I will ping you if i need some help

**Comment by fondberg** [comment]
Is this still an issue?

**Comment by IAmBecomeDeth** [comment]
I can confirm this as well. I came to report it. 

Functionally works, but once playing music, shows wrong cast destination. I have 12 cast devices and 5 or so groups for each room as well as multiple "party" settings for including or not including bedrooms or bathrooms. I'm not sure if I'm overloading the list, maybe? 

When I choose to play on "party" or "party2 or any cast group (99% of usage) it says "Bedroom Second Speaker". If I choose a specific device, it will say the correct device. If I set cast to party from my phone, current player shows correctly in both Spotify app and Lovelace.

I noticed that when selecting from phone, it shows "Party" (chrome cast group) under Spotify connect. When listed under chrome cast it does not show correctly.

**Comment by IAmBecomeDeth** [comment]
Testing more over the past few hours... The incorrect device that's being displayed is not static. Now it shows a different (also incorrect) device. Additionally, I figured out that the current destination shows up as Spotify connect whether correct or incorrect, so that isn't the reason. I'm going to test adding party to my known devices and see if it makes any difference. Won't be able to test til tomorrow.

It's really odd that it's showing the information incorrectly in Spotify app...yet still working as expected on the correct speaker group.
